micrometer-core/src/main/java/io/micrometer/core/instrument/step/StepMeterRegistry.java

```diff
@@ -116,4 +142,22 @@ public void close() {
         super.close();
     }
 
+    /**
+     * This will poll the values from meters, which will cause a roll over for Step-meters
+     * if past the step boundary. This gives some control over when roll over happens
+     * separate from when publishing happens.
+     */
+    // VisibleForTesting
+    void pollMetersToRollover() {
+        this.getMeters()
+            .forEach(m -> m.match(gauge -> null, Counter::count, Timer::count, DistributionSummary::count,
+                    meter -> null, meter -> null, FunctionCounter::count, FunctionTimer::count, meter -> null));
+    }
+
+    private long getInitialDelay() {
+        long stepMillis = config.step().toMillis();
+        // schedule one millisecond into the next step
+        return stepMillis - clock.wallTime() % stepMillis + 1;
```

Comment:
Can we make it more obvious?
```suggestion
        return stepMillis - (clock.wallTime() % stepMillis) + 1;
```